core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -231,6 +216,24 @@ private Object handleTxCommand(TxInvocationContext ctx, TransactionBoundaryComma
          }
       } else {
          if (currentTopologyId() > command.getTopologyId()) {
+            /*
+             * We need to retry in case of topology-id mismatch.
+             * We rollback the transaction is it is a 2PC. The rollback is safe (even with state transfer prepared transactions)
+            because we ask the coordinator
+            to retry the transaction. The retry will re-acquire the locks and perform the validation needed. If this node
```

Comment:
only now I saw the JIRA. Well, +1 point for me. the JIRA is already solved and I don't break any existing tests.